test/runtests.jl

```diff
@@ -16,12 +17,13 @@ end
 @testset "Mocking" begin
     include("compiled-modules.jl")
     include("expr.jl")
+    include("dispatch.jl")
     include("bindings/bindings.jl")
-    include("patch.jl")
+    # include("patch.jl")
```

Comment:
why is commented out?

Reply:
Those tests need to be removed and no longer work. I avoided removing it here as there are allot of changes already.

Reply:
What is being made easier by commenting them out rather than deleting them?

Reply:
By just commenting this single line out you don't have to also see the diff of the deleted patch.jl test file. I can include that change here as well since you are now familiar with these changes.

Reply:
please include them here.

Reply:
Looking over the patch tests again I added some new tests for specifically testing the `@patch` macro. So now this file is updated instead of just removed.